internal/pkg/policy/monitor.go

```diff
@@ -96,72 +105,88 @@ func NewMonitor(bulker bulk.Bulk, monitor monitor.Monitor, throttle time.Duratio
 
 // Run runs the monitor.
 func (m *monitorT) Run(ctx context.Context) error {
-	m.log.Info().Dur("throttle", m.throttle).Msg("run policy monitor")
+	m.log.Info().
+		Dur("throttle", m.throttle).
+		Msg("run policy monitor")
 
 	s := m.monitor.Subscribe()
 	defer m.monitor.Unsubscribe(s)
 
-	// The output from m.monitor times out if we don't pull data off quickly enough.
-	// Rollout can take a while; append upates here until we can attend to it.
-	// TODO: This is a workaround for 7.14.  Rollout strategy will be reconsidered for 7.15.
-	subCtx, cfunc := context.WithCancel(ctx)
-	defer cfunc()
-	outputCh := m.monitorOutputChannel(subCtx, s.Output())
+	// If no throttle set, setup a minimal spin rate.
+	dur := m.throttle
+	if dur == 0 {
+		dur = time.Nanosecond
+	}
+
+	isDeploying := true
+	ticker := time.NewTicker(dur)
+	defer ticker.Stop()
+
+	startDeploy := func() {
+		if !isDeploying {
+			isDeploying = true
+			ticker = time.NewTicker(dur)
+		}
+	}
+
+	stopDeploy := func() {
+		ticker.Stop()
+		isDeploying = false
+	}
+
+	// begin in stopped state
+	stopDeploy()
 
 	close(m.startCh)
+
 LOOP:
 	for {
 		select {
-		case <-ctx.Done():
-			break LOOP
 		case <-m.kickCh:
-			if err := m.process(ctx); err != nil {
+			if err := m.loadPolicies(ctx); err != nil {
 				return err
 			}
-		case hits := <-outputCh:
-			policies := make([]model.Policy, len(hits))
-			for i, hit := range hits {
-				err := hit.Unmarshal(&policies[i])
-				if err != nil {
-					return err
-				}
-			}
-			if err := m.processPolicies(ctx, policies); err != nil {
+			startDeploy()
+		case <-m.deployCh:
+			startDeploy()
+		case hits := <-s.Output():
+			if err := m.processHits(ctx, hits); err != nil {
 				return err
 			}
+			startDeploy()
+		case <-ticker.C:
+			if done := m.dispatchPending(); done {
+				stopDeploy()
+			}
+		case <-ctx.Done():
+			break LOOP
```

Comment:
this will leak the ticker if it was started by any of the previous ```case```s

Reply:
whoops fixed.